### PR TITLE
Add validations and unit test for all the models

### PR DESCRIPTION
**Pull request by kennedyowusu**
# Highlights of this PR

 - [x] Add the validations for the User model

- [ ] Name must not be blank.
- [ ] Posts Counter must be an integer greater than or equal to zero.

- [x] Add the validations for the Post model

- [ ] Title must not be blank.
- [ ] Title must not exceed 250 characters.
- [ ] Comments Counter must be an integer greater than or equal to zero.
- [ ] Likes Counter must be an integer greater than or equal to zero.

- [x] Add unit specs for all of the models' methods and validations.

Below is a short of the result of the testing of all the models

test-result-for-all-models

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -2,8 +2,7 @@
```
Comment (arslanbisharat) [inline spec/models/comment_spec.rb]: - [ ] In your project, you have file with '0' as the name. Please update it so it looks professional. 

image

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @kennedyowusu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The Pull Request has a proper title and description. ✔️
- You have added a descriptive Readme file. Good Job. ✔️
- There is no linter error is present. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Please make sure you right descriptive comments. Commenting involves placing Human Readable Descriptions inside of computer programs detailing what the Code is doing. Proper use of commenting can make code maintenance much easier, as well as helping make finding bugs faster. Further, commenting is very important when writing functions that other people will use. :x:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (kennedyowusu) [reply spec/models/comment_spec.rb]: Thank you for reviewing the project. This has been resolved

**Review by jasonfelice — APPROVED** [review]
Hi @kennedyowusu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<p align="center"></p>

## Highlights
Descriptive PR✔️
No Linter errors✔️
Professional Documentation✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kennedyowusu** [comment]
Thank you @jasonfelice for taking the time to review and approve the project.

**Pull request merged**